src/snowflake/snowpark/functions.py

```diff
@@ -11064,3 +11148,138 @@ def try_to_binary(
+
+    # Call the built-in Base64 encode function.
+    return builtin("base64_encode", _emit_ast=_emit_ast)(*args)
+
+
+base64 = base64_encode
+
+
+@publicapi
+def base64_decode_string(
+    e: ColumnOrName, alphabet: Optional[str] = None, _emit_ast: bool = True
+) -> Column:
+    """
+    Decodes a Base64-encoded string to a string.
+
+    Example:
+        >>> df = session.create_dataframe(["U25vd2ZsYWtl", "SEVMTE8="], schema=["input"])
+        >>> df.select(base64_decode_string(col("input")).alias("decoded")).collect()
+        [Row(DECODED='Snowflake'), Row(DECODED='HELLO')]
+    """
+    # Convert input to a column if it is not already one.
+    col_input = _to_col_if_str(e, "base64_decode_string")
+
+    # Prepare arguments for the function call.
+    args = [col_input]
+
+    if alphabet:
+        args.append(lit(alphabet))
+
+    # Call the built-in Base64 encode function.
+    return builtin("base64_decode_string", _emit_ast=_emit_ast)(*args)
+
+
+unbase64 = base64_decode_string
+
+
+@publicapi
+def hex_encode(e: ColumnOrName, case: int = 1, _emit_ast: bool = True):
+    """
+    Encodes the input using hexadecimal (also ‘hex’ or ‘base16’) encoding.
+
+    Example:
+        >>> df = session.create_dataframe(["Snowflake", "Hello"], schema=["input"])
+        >>> df.select(hex_encode(col("input")).alias("hex_encoded")).collect()
+        [Row(HEX_ENCODED='536E6F77666C616B65'), Row(HEX_ENCODED='48656C6C6F')]
+    """
+    col_input = _to_col_if_str(e, "base64_decode_string")
+    return builtin("hex_encode", _emit_ast=_emit_ast)(col_input, lit(case))
+
+
+hex = hex_encode
+
+
+@publicapi
+def editdistance(
+    string_expr1: ColumnOrName,
+    string_expr2: ColumnOrName,
+    max_distance: Optional[int] = None,
+    _emit_ast: bool = True,
+) -> Column:
+    """Computes the Levenshtein distance between two input strings.
+
+    Optionally, a maximum distance can be specified. If the distance exceeds this value,
+    the computation halts and returns the maximum distance.
+
+    Example::
+
+        >>> df = session.create_dataframe(
+        ...     [["abc", "def"], ["abcdef", "abc"], ["snow", "flake"]],
+        ...     schema=["s1", "s2"]
+        ... )
+        >>> df.select(
+        ...     editdistance(col("s1"), col("s2")).alias("distance"),
+        ...     editdistance(col("s1"), col("s2"), 2).alias("max_2_distance")
+        ... ).collect()
+        [Row(DISTANCE=3, MAX_2_DISTANCE=2), Row(DISTANCE=3, MAX_2_DISTANCE=2), Row(DISTANCE=5, MAX_2_DISTANCE=2)]
+    """
+    s1 = _to_col_if_str(string_expr1, "editdistance")
+    s2 = _to_col_if_str(string_expr2, "editdistance")
+
+    args = [s1, s2]
+    if max_distance is not None:
+        max_dist = (
+            max_distance
+            if isinstance(max_distance, Column)
+            else lit(max_distance, _emit_ast=_emit_ast)
+        )
+        args.append(max_dist)
+
+    return builtin("editdistance", _emit_ast=False)(*args)
+
+
+levenshtein = editdistance
+
```

Comment:
from SAS perspective this could create a problem for example

in snowflake 
```
SELECT EDITDISTANCE('kitten', 'sitting', 2) ;
2
```
but in pyspark it is -1. see https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.levenshtein.html

Reply:
that's a good point.
IMHO I think snowpark should be consistent with snowflake's behavior, and fixing the inconsistency is SAS's work.

Reply:
makes sense. How about not only keeping `editdistance` which is supported on the server side removing `levenshtein` to reducing confusion?

Reply:
sounds like a good plan to me